modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/Slot.java

```diff
@@ -109,6 +149,64 @@ public void setDestinationOfMessagesInSlot(String destinationOfMessagesInSlot) {
         this.destinationOfMessagesInSlot = destinationOfMessagesInSlot;
     }
 
+    /**
+     * Check if state going to be added is valid considering it as the next
+     * transition compared to current latest state.
+     * @param state state to be transferred
+     */
+    public boolean addState(SlotState state) {
+
+        boolean isValidTransition = false;
+
+        if(slotStates.isEmpty()) {
+            if(SlotState.CREATED.equals(state)) {
+                isValidTransition = true;
+                slotStates.add(state);
+            } else {
+                log.warn("Invalid State transition suggested: " + state);
+            }
+        } else {
+            isValidTransition = slotStates.get(slotStates.size() - 1).isValidNextTransition(state);
+            if(isValidTransition) {
+                slotStates.add(state);
+            } else {
+                log.warn("Invalid State transition from " + slotStates.get
+                        (slotStates.size() - 1) + " suggested: " + state + " Slot ID: " + this
+                        .getId());
+
+            }
+        }
+
+        return isValidTransition;
+    }
+
+    /**
+     * Convert Slot state list to a string
+     * @return Encoded string
+     */
+    public String encodeSlotStates() {
+        String encodedString;
+        StringBuilder builder = new StringBuilder();
+        for (SlotState slotState : slotStates) {
+            builder.append(slotState.getCode()).append("%");
+        }
+        encodedString = builder.toString();
```

Comment:
what about just return `builder.toString()`? Get rid of the `encodedString` variable